src/core/dbus.c

```diff
@@ -1166,15 +1167,33 @@ int bus_foreach_bus(
                 RET_GATHER(r, send_message(b, userdata));
         }
 
-        /* Send to API bus, but only if somebody is subscribed */
+        /* Send to API bus, if have subscribers and if we should honor them */
         if (m->api_bus &&
-            (sd_bus_track_count(m->subscribed) > 0 ||
-             sd_bus_track_count(subscribed2) > 0))
+                (!only_with_subscribers ||
+                 sd_bus_track_count(subscribed) > 0))
                 RET_GATHER(r, send_message(m->api_bus, userdata));
 
         return r;
 }
 
+int bus_foreach_bus_to_all(Manager *m,
+                           int (*send_message)(sd_bus *bus, void *userdata),
+                           void *userdata)
+{
+        return bus_foreach_bus_internal(m, NULL, false, send_message, userdata);
+}
+
+int bus_foreach_bus(
+                Manager *m,
+                sd_bus_track *subscribed2,
+                int (*send_message)(sd_bus *bus, void *userdata),
+                void *userdata) {
+        sd_bus_track *subscribed;
+
+        subscribed = sd_bus_track_count(subscribed2) > 0 ? subscribed2 : m->subscribed;
```

Comment:
This seems spurious. The logic for `m->subscribed` should very much be left in bus_foreach_bus_internal(). With your change we suddenly stop sending out signals for clients subscribed to Manager if someone's subscribed to single object (job/unit)

Reply:
Is it? In such case we still do the same action of before, otherwise we just follow the global subscribed check.